src/box/lua/config/applier/credentials.lua

```diff
@@ -895,13 +952,20 @@ local function apply(config_module)
         wait_sync = fiber.channel()
     end
 
+    if not schema_is_upgraded_cond then
+        schema_is_upgraded_cond = fiber.cond()
+    end
+    update_schema_upgraded_status()
+
     -- Set trigger on after space/function/sequence creation.
     if not triggers_are_set then
         triggers_are_set = true
+
         box.space._user:on_replace(on_replace_trigger)
         box.space._space:on_replace(on_replace_trigger)
         box.space._func:on_replace(on_replace_trigger)
         box.space._sequence:on_replace(on_replace_trigger)
+        box.space._schema:on_replace(credentials_upgrade_on_replace)
```

Comment:
Nit: we don't need to set that trigger, if `schema_is_upgraded = true`

Reply:
Done! Added an extra variable to check if the trigger is set and set it only if the trigger is unset and the schema wasn't upgraded. Thank you!